query/src/main/java/org/infinispan/query/logging/Log.java

```diff
@@ -94,4 +95,28 @@
    @LogMessage(level = ERROR)
    @Message(value = "Error executing MassIndexer", id = 14018)
    void errorExecutingMassIndexer(@Cause Throwable cause);
+
+   @Message(value = "Cannot run Lucene queries on a cache that does not have indexing enabled", id = 14019)
+   IllegalStateException cannotRunLuceneQueriesIfNotIndexed();
+
+   @Message(value = "Query parameter '%s' was not set", id = 14020)
+   IllegalStateException queryParameterNotSet(String paramName);
+
+   @Message(value = "Queries containing grouping and aggregation functions must use projections.",  id = 14021)
+   ParsingException groupingAndAggregationQueriesMustUseProjections();
+
+   @Message(value = "Cannot have aggregate functions in GROUP BY clause", id = 14022)
+   IllegalStateException cannotHaveAggregationsInGroupByClause();
+
+   @Message(value = "The property path '%s' cannot be used in the GROUP BY clause because it is multi-valued", id = 14023)
+   ParsingException multivaluedPropertyCannotBeUsedInGroupBy(String propertyPath);
```

Comment:
It'd be nicer to have a message like "not currently supported", since it could be added later